### PR TITLE
core: Add option for increasing socket send buffer size

### PR DESCRIPTION
**Pull request by btriller**
#### Pre-Submission Checklist

- [ ] Commit message has the format required by CONTRIBUTING guide
- [ ] Commits are split per component (core, individual modules, libs, utils, ...)
- [ ] Each component has a single commit (if not, squash them into one commit)
- [ ] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [ ] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

### CODE REVIEW

**Comment by miconda** [comment]
The diff has a lot of formatting changes, did you run `clang-format` or you did them manually?

New features should be made as a clean PR, without formatting changes, because it is hard to follow up what is new and what is reformatting, you also relocated some lines, adding to the volume of the changes.

Reformatting PR's are also welcome, but they have to be done separately, so one knows it is not about behavioural changes. But not that clang-format should be used at the end to ensure coherency.

Can you split the PR in two, one with the new feature and the second with formatting/relocating changes?

**Comment by btriller** [comment]
Did a `clang-format` after `check-format` failed. Force pushed w/o format changes.

**Comment by miconda** [comment]
Thanks! Merging this one.

You can make a new PR with the format changes.
